rclrs_tests/src/parameter_service_tests.rs

```diff
@@ -0,0 +1,618 @@
+    let node = NodeBuilder::new(context, "node")
+        .namespace(ns)
+        .build()
+        .unwrap();
+    let range = ParameterRange {
+        lower: Some(0),
+        upper: Some(100),
+        step: None,
+    };
+    let bool_param = node
+        .declare_parameter("bool")
+        .default(true)
+        .description("A boolean value")
+        .mandatory()
+        .unwrap();
+    let _ns_param = node
+        .declare_parameter("ns1.ns2.ns3.int")
+        .default(42)
+        .range(range)
+        .constraints("Only the answer")
+        .mandatory()
+        .unwrap();
+    let _read_only_param = node
+        .declare_parameter("read_only")
+        .default(1.23)
+        .read_only()
+        .unwrap();
+    let dynamic_param = node
+        .declare_parameter("dynamic")
+        .default(ParameterValue::String("hello".into()))
+        .mandatory()
+        .unwrap();
+
+    let client = NodeBuilder::new(context, "client")
+        .namespace(ns)
+        .build()
+        .unwrap();
+
+    (
+        TestNode {
+            node,
+            bool_param,
+            _ns_param,
+            _read_only_param,
+            dynamic_param,
+        },
+        client,
+    )
+}
+
+#[test]
+fn test_parameter_services_names_and_types() -> Result<(), RclrsError> {
+    let context = Context::new([]).unwrap();
+    let (node, _client) = construct_test_nodes(&context, "names_types");
+    let check_names_and_types = |names_and_types: TopicNamesAndTypes| {
+        let types = names_and_types
+            .get("/names_types/node/describe_parameters")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/DescribeParameters".to_string()));
+        let types = names_and_types
+            .get("/names_types/node/get_parameters")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/GetParameters".to_string()));
+        let types = names_and_types
+            .get("/names_types/node/set_parameters")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/SetParameters".to_string()));
+        let types = names_and_types
+            .get("/names_types/node/set_parameters_atomically")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/SetParametersAtomically".to_string()));
+        let types = names_and_types
+            .get("/names_types/node/list_parameters")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/ListParameters".to_string()));
+        let types = names_and_types
+            .get("/names_types/node/get_parameter_types")
+            .unwrap();
+        assert!(types.contains(&"rcl_interfaces/srv/GetParameterTypes".to_string()));
+    };
+
+    std::thread::sleep(std::time::Duration::from_millis(100));
+
+    let service_names_and_types = node.node.get_service_names_and_types()?;
+    check_names_and_types(service_names_and_types);
```

Comment:
I'm very confused about why we put all the tests into a closure up above and then call that closure only once here. I would understand using a closure if we called it multiple times, but we don't seem to do that.

It's not wrong, but it leaves me wondering if I'm misunderstanding something about the test.

Reply:
I feel at some point of the writing the closure was reused, but I agree it doesn't make any sense now. https://github.com/ros2-rust/ros2_rust/pull/342/commits/a7d1befbe078418fca4852b200e3e1b6c4a62f9f